src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -613,3 +655,22 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
         }
     }
 }
+
+static class RandomExtensions
+{
+
+    public static string GetRandomSubstring(this Random random, string input)
+    {
+        var offset = random.Next(0, input.Length);
+        var length = random.Next(0, input.Length - offset + 1);
+        return input.Substring(offset, length);
+    }
+
+    public static bool NextBoolean(this Random random, double probability)
+    {
+        if (probability < 0 || probability > 1)
+            throw new ArgumentOutOfRangeException("probability", "must be between 0 and 1");
+
+        return random.NextDouble() < probability;
+    }
```

Comment:
Seems like these could just be instance methods on ClientContext and make Random private.  I don't think having these be extension methods adds anything here.

Reply:
I intentionally segregated behaviour from data in the ClientContext. Not too religious about it, can integrate if it's an issue.